src/borg/archive.py

```diff
@@ -1401,32 +1397,45 @@ def check(self, repository, repair=False, archive=None, first=0, last=0, sort_by
         :param verify_data: integrity verification of data referenced by archives
         :param save_space: Repository.commit(save_space)
         """
-        logger.info('Starting archive consistency check...')
-        self.check_all = archive is None and not any((first, last, glob))
+        self.error_found = False
+        self.verify_data = verify_data
+        self.possibly_superseded = set()
         self.repair = repair
         self.repository = repository
+        self.archive = archive
+        self.first = first
+        self.last = last
+        self.glob = glob
+        self.check_all = archive is None and not any((first, last, glob))
+        self.sort_by = sort_by
+        self.save_space = save_space
         self.init_chunks()
+        self.key = self.identify_key(repository)
```

Comment:
guess these 2 need checking.

do these have any unwanted consequences when run before the repo check has happened?